range_queries/prefix_sum_array.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * @file prefix_sum_array.cpp
+ * @brief Implementation of [Prefix Sum Array] data structure
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂